pkg/resources/network_policy_acceptance_test.go

```diff
@@ -0,0 +1,55 @@
+package resources_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+const (
+	networkPolicyComment = "Created by a Terraform acceptance test"
+)
+
+func TestAccNetworkPolicy(t *testing.T) {
+	if _, ok := os.LookupEnv("SKIP_NETWORK_POLICY_TESTS"); ok {
+		t.Skip("Skipping TestAccNetworkPolicy")
+	}
+
+	name := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+
+	resource.Test(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: networkPolicyConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_network_policy.test", "name", name),
+					resource.TestCheckResourceAttr("snowflake_network_policy.test", "comment", networkPolicyComment),
+					resource.TestCheckResourceAttr("snowflake_network_policy.test", "allowed_ip_list.#", "2"),
+				),
+			},
+			// TODO (gp) figure out how to test UPDATE here
+			// IMPORT
+			{
+				ResourceName:            "snowflake_network_policy.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"set_for_account"},
+			},
+		},
+	})
+}
+
+func networkPolicyConfig(name string) string {
+	return fmt.Sprintf(`
+resource "snowflake_network_policy" "test" {
+	name            = "%v"
+	comment         = "%v"
+	allowed_ip_list = ["192.168.0.100/24", "29.254.123.20"]
+	blocked_ip_list = ["192.168.0.101"]
```

Comment:
We should probably include testing adding to users here.

Reply:
I've added a `users` list to the test network policy resource definition (and a dummy user resource definition) so at least the user-granting code should be executing during this acceptance test now.

However, (as far as I can tell) I have to define:
`ImportStateVerifyIgnore: []string{"set_for_account", "users"},` 

because nowhere in `ReadNetworkPolicy` is there:
`data.Set("set_for_account", ...)` or
`data.Set("users", ...)`

This is the case because there doesn't appear to be a way in Snowflake to do "get all users with this Network Policy set" or "tell me if this network policy is set for the Snowflake account". Neither `SHOW` nor `DESC` returns any information about the active/applied status of a given Network Policy:

- Show: https://docs.snowflake.com/en/sql-reference/sql/show-network-policies.html
- Describe: https://docs.snowflake.com/en/sql-reference/sql/desc-network-policy.html

Is it acceptable to leave those out of `ReadNetworkPolicy` (and thus having to include them in `ImportStateVerifyIgnore`)?

Reply:
FWIW this is why I had to write this resource with `ForceNew: true` for those two fields, with the "exclusive assignment" explanation in the docs — the simplest way I could figure out how to deal with this is to just always destroy/recreate and re-grant to users or re-set on account

Reply:
Ah that makes sense!

Thinking about this a bit more, I think we should consider separating the attachment to a separate resource. This should allow us to avoid the scenario where we need to recreate the network policy resource every time one of the attachments changes. Of course the attachments may need to be recreated, but that might be tolerable. What do you think?

Reply:
I'm on board with this — a separate attachment resource was my original plan, but I moved away from it because I (erroneously) observed that if you dropped a network policy, the users with that policy set continued to be restricted by the policy. Hence, I thought we needed to keep the attachment tightly coupled with the resource. 

However, I tested this understanding again and it turns out there's just a bit of lag (not more than a minute) between dropping the network policy and the user then being able to open a session from any IP address

So this approach _should_ work. I'll update PR

Reply:
I guess one thing to think about is that if we do this, the `snowflake_network_policy_attachment` resource wouldn't behave like the similar *_grant resources (e.g. warehouse/database/etc) and create exclusive attachments

Because there's not a way of getting existing attachments from `SHOW`/`DESC`ing network policies/users/accounts, a network policy could be attached to a user outside of Terraform, and then changes to the `snowflake_network_policy_attachment` wouldn't destroy that attachment. That's probably fine? We should just document it

Reply:
Yeah I think that is fine given the limitations we are working with.

Reply:
Done — the one thing I'm unsure about is `ReadNetworkPolicyAttachment` — given above limitations, there's no way to actually implement `schema.ReadFunc` for the attachment resource. I don't think I actually need `ReadNetworkPolicyAttachment` implemented for this, but `InternalValidate` fails without it. To get around that, I just implemented a dummy func 🤷 

lmk if there's a better way to handle this!